service/collector/vpn_connection.go

```diff
@@ -131,7 +126,11 @@ func (v *VPNConnection) Describe(ch chan<- *prometheus.Desc) error {
 }
 
 func (v *VPNConnection) getVPNConnectionsClient() (*network.VirtualNetworkGatewayConnectionsClient, error) {
-	azureClients, err := client.NewAzureClientSet(v.hostAzureClientSetConfig)
+	config, err := credential.GetAzureConfigFromSecretName(v.k8sClient, credential.CredentialDefault, credential.CredentialNamespace)
```

Comment:
This is probably the biggest change.

Before this PR, credentials were passed to the operator (now collector). These credentials were credentials to be able to talk to the control plane subscription.
With this change, no credentials are passed. Instead, the credentials stored in the `credential-default` secret are used. Does this look right @giantswarm/team-celestial ?

Reply:
I think so, yeah.